### DISCUSSION TITLE
Investments with technology degredation

### DISCUSSION THREAD

**Discussion by mihlema** [spine-tools/SpineOpt.jl#865] [Ideas]
This issue addresses the missing functionality of mimicking the degradation process of investments. Currently, we only account for an economic lifetime of investments, i.e. how long until a unit has paid for itself (@DillonJ please correct me if this is incorrect). 

The technical lifetime is not represented, meaning that a unit once installed can be used forever. Moreover, the performance of a unit does not change over time. To account for aging processes we should consider taking this into account.
Missing functionality in this regard is hence:
- [ ] decommissioning of units after their technical lifetime
- [ ] continuous degradation of the efficiency over time
- [ ] (mothballing of technology - not now)

The first point should be quite straight forward, similar to the current implementation of `add_constraint_unit_lifetime!` (which should be renamed to economic lifetime), but would enforce an upper bound on `units_invested_available`.
But we should also be careful about units that were available at the start of the optimization,i.e. `number_of_units`, and made sure we are able to decomission them.

The second point is slightly more tricky. One option could be to have a constant factor, giving the efficiency decrease percentage per year, e.g. 1%/year <sup>1</sup>. This degradation factor would be multiplied with the `fix_ratio_out_in`, `max_ratio_out_in` (per year after installation). For a non-constant degradation factor e.g. `[year1: 1%, year2: 2% ...]`, I think we would need a mapping parameter of the form `duration -> value`, but we don't have support for this from SpineInterface. 

Edit: We do have support from SpineInterface for this, using the mapping parameter. I wonder if this mapping would work in combination with stochastic scenarios?
<sup>1</sup> Arguable [year] unit, but that's another discussion: #82

**Comment by clizbe** [comment]
Closing due to stale status.
